server/state/error.go

```diff
@@ -0,0 +1,33 @@
+package state
+
+import (
+	"fmt"
+	"strings"
+)
+
+// Reports that no sandbox with given ID exists
+func errNoSuchSandbox(id string) error {
+	return fmt.Errorf("no sandbox found with ID %s", id)
+}
+
+// Reports that no container with given ID exists
+func errNoSuchContainer(id string) error {
+	return fmt.Errorf("no container found with ID %s", id)
+}
+
+// Reports that no container with given ID exists in given sandbox
+func errNoSuchContainerInSandbox(id string, sandboxID string) error {
+	return fmt.Errorf("no container found with ID %s in sandbox %s", id, sandboxID)
+}
+
+// Functions for verifying errors
+
+// IsSandboxNotExist checks if an error indicated that given sandbox does not exist
+func IsSandboxNotExist(err error) bool {
+	return strings.HasPrefix(err.Error(), "no sandbox found with ID")
+}
```

Comment:
This is a kind of weird way to check error types. Take a look at e.g. api/server/router/container/container.go validationError for a nicer way that uses types instead of string comparisons.